File: Backend/OhDeerBackend/OhDeerBackend/Models/Fee.cs
using OhDeerBackend.Models.Base_Classes;
using System.ComponentModel.DataAnnotations;

namespace OhDeerBackend.Models
{
    public class Fee : BaseEntity
    {
        public bool IsPaid { get; set; }

        [Required]
        public DateTime DueDate { get; set; }

        [Required]
        public int Amount { get; set; }

        // Navigation Properties
        public virtual ICollection<Purchase> Purchases { get; set; } = new List<Purchase>();
    }

}

File: Backend/app/Models/Fee.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Fee extends Model
{
    use HasFactory;

    protected $fillable = ['IsPaid', 'DueDate', 'Amount'];

    protected $casts = [
        'IsPaid' => 'boolean',
        'DueDate' => 'datetime',
    ];

    public function purchases()
    {
        return $this->hasMany(Purchase::class);
    }
}
